src/query/interpreter.cpp

```diff
@@ -1909,6 +1918,119 @@ PreparedQuery PrepareSettingQuery(ParsedQuery parsed_query, const bool in_explic
   // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
 }
 
+Callback HandleTransactionQueueQuery(TransactionQueueQuery *transaction_query,
+                                     const std::optional<std::string> &username, uint64_t current_transaction_id,
+                                     const Parameters &parameters, InterpreterContext *interpreter_context,
+                                     DbAccessor *db_accessor) {
+  Frame frame(0);
+  SymbolTable symbol_table;
+  EvaluationContext evaluation_context;
+  evaluation_context.timestamp = QueryTimestamp();
+  evaluation_context.parameters = parameters;
+  ExpressionEvaluator evaluator(&frame, symbol_table, evaluation_context, db_accessor, storage::View::OLD);
+
+  bool isAdmin = interpreter_context->auth_checker->IsUserAdmin(username);
+
+  Callback callback;
+  switch (transaction_query->action_) {
+    case TransactionQueueQuery::Action::SHOW_TRANSACTIONS: {
+      callback.header = {"username", "transaction_id", "queries summary"};
+      callback.fn = [interpreter_context, username, isAdmin]() mutable {
+        std::vector<std::vector<TypedValue>> results;
+        results.reserve(interpreter_context->interpreters->size());
+        interpreter_context->interpreters.WithLock([&results, username, isAdmin](const auto &interpreters) {
+          spdlog::info("Number of interpreters in callback: {}", interpreters.size());
+          for (Interpreter *interpreter : interpreters) {
+            if (interpreter->HasActiveTransaction() && (interpreter->username_ == username || isAdmin)) {
+              auto queries_summaries_typed = std::vector<TypedValue>();
+              auto queries_summaries = interpreter->GetQueriesSummary();  // in std::transform
+              std::for_each(queries_summaries.begin(), queries_summaries.end(),
+                            [&queries_summaries_typed](const auto &query_summary) {
+                              queries_summaries_typed.emplace_back(query_summary);
+                            });
+
+              results.push_back({TypedValue(interpreter->username_.value_or("")),
+                                 TypedValue(std::to_string(interpreter->GetTransactionId())),
+                                 TypedValue(queries_summaries_typed)});
+            }
+          }
+        });
+        return results;
+      };
+      break;
+    }
+    case TransactionQueueQuery::Action::TERMINATE_TRANSACTIONS: {
+      std::vector<std::string> transaction_ids;
```

Comment:
I would rename this variable to something more descriptive. User would like to kill those transactions, if they are found and if he has rights, so maybe:

kill_transaction_ids
maybe_kill_transaction_ids

Reply:
fico linguista, I like it! will do the change